test/common/creation.go

```diff
@@ -17,47 +17,67 @@ limitations under the License.
 package common
 
 import (
-	eventingv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
-	sourcesv1alpha1 "github.com/knative/eventing/pkg/apis/sources/v1alpha1"
 	"github.com/knative/eventing/test/base"
 	corev1 "k8s.io/api/core/v1"
 	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// TODO(Fredy-Z): break this file into multiple files when it grows too large.
+
 var coreAPIGroup = corev1.SchemeGroupVersion.Group
 var coreAPIVersion = corev1.SchemeGroupVersion.Version
 var rbacAPIGroup = rbacv1.SchemeGroupVersion.Group
 var rbacAPIVersion = rbacv1.SchemeGroupVersion.Version
 
-// CreateChannelOrFail will create a Channel or fail the test if there is an error.
-func (client *Client) CreateChannelOrFail(name, provisonerName string) {
-	namespace := client.Namespace
-	channel := base.Channel(name, provisonerName)
+// CreateChannelOrFail will create a Channel Resource in Eventing.
+// TODO(Fredy-Z): This is a workaround when there are both provisioner and Channel CRDs in this repo.
+//                isCRD needs to be deleted when the provisioner implementation is removed.
+func (client *Client) CreateChannelOrFail(name string, channelTypeMeta *metav1.TypeMeta, provisionerName string) {
+	if channelTypeMeta.Kind == base.ChannelKind {
+		namespace := client.Namespace
+		channel := base.Channel(name, provisionerName)
+
+		channels := client.Eventing.EventingV1alpha1().Channels(namespace)
+		// update channel with the new reference
+		channel, err := channels.Create(channel)
+		if err != nil {
+			client.T.Fatalf("Failed to create channel %q: %v", name, err)
+		}
+		client.Cleaner.AddObj(channel)
+	} else {
+		client.createChannelCROrFail(name, channelTypeMeta)
+	}
+}
 
-	channels := client.Eventing.EventingV1alpha1().Channels(namespace)
-	// update channel with the new reference
-	channel, err := channels.Create(channel)
-	if err != nil {
-		client.T.Fatalf("Failed to create channel %q: %v", name, err)
+func (client *Client) createChannelCROrFail(name string, channelTypeMeta *metav1.TypeMeta) {
+	namespace := client.Namespace
+	if channelTypeMeta.Kind == base.KafkaChannelKind {
+		channel := base.KafkaChannel(name)
```

Comment:
set the namespace

Reply:
Explained in another comment, we do not need to set namespace here.